digiwf-task/digiwf-tasklist-service/src/test/java/io/muenchendigital/digiwf/task/service/adapter/out/engine/RemoteTaskCommandRestAdapterTest.java

```diff
@@ -6,56 +6,79 @@
 
 class RemoteTaskCommandRestAdapterTest {
-  private final TaskService taskService = mock(TaskService.class);
-  private final TaskCommandPort taskCommandPort = new RemoteTaskCommandRestAdapter(taskService);
-  private final String taskId = UUID.randomUUID().toString();
-
-  private final VariableFactory<String> STRING_VAR = stringVariable("STRING_VAR");
-
-  @Test
-  public void cancels_task() {
-    taskCommandPort.cancelUserTask(taskId);
-    verify(taskService).handleBpmnError(taskId, BpmnErrors.DEFAULT_TASK_CANCELLATION_ERROR);
-  }
-
-  @Test
-  public void assign() {
-    taskCommandPort.assignUserTask(taskId, "user");
-    verify(taskService).setAssignee(taskId, "user");
-  }
-
-  @Test
-  public void unassign() {
-    taskCommandPort.unassignUserTask(taskId);
-    verify(taskService).setAssignee(taskId, null);
-  }
-
-  @Test
-  public void complete() {
-    val vars = CamundaBpmData
-        .builder()
-        .set(STRING_VAR, "init")
-        .build();
-    taskCommandPort.completeUserTask(taskId, vars);
-    verify(taskService).complete(taskId, vars);
-  }
-
-  @Test
-  public void save() {
-    val vars = CamundaBpmData
-        .builder()
-        .set(STRING_VAR, "init")
-        .build();
-    taskCommandPort.saveUserTask(taskId, vars);
-    verify(taskService).setVariables(taskId, vars);
-  }
-}
\ No newline at end of file
+    private final TaskService taskService = mock(TaskService.class);
+    private final TaskCommandPort taskCommandPort = new RemoteTaskCommandRestAdapter(taskService);
+    private final String taskId = UUID.randomUUID().toString();
+
+    private final VariableFactory<String> STRING_VAR = stringVariable("STRING_VAR");
+
+    @Test
+    public void cancels_task() {
+        taskCommandPort.cancelUserTask(taskId);
+        verify(taskService).handleBpmnError(taskId, BpmnErrors.DEFAULT_TASK_CANCELLATION_ERROR);
+    }
+
+    @Test
+    public void assign() {
+        taskCommandPort.assignUserTask(taskId, "user");
+        verify(taskService).setAssignee(taskId, "user");
+    }
+
+    @Test
+    public void unassign() {
+        taskCommandPort.unassignUserTask(taskId);
+        verify(taskService).setAssignee(taskId, null);
+    }
+
+    @Test
+    public void complete() {
+        val vars = CamundaBpmData
+                .builder()
+                .set(STRING_VAR, "init")
+                .build();
+        taskCommandPort.completeUserTask(taskId, vars);
+        verify(taskService).complete(taskId, vars);
+    }
+
+    @Test
+    public void save() {
+        val vars = CamundaBpmData
+                .builder()
+                .set(STRING_VAR, "init")
+                .build();
+        taskCommandPort.saveUserTask(taskId, vars);
+        verify(taskService).setVariables(taskId, vars);
+    }
+
+    @Test
+    public void deferUserTask() {
+        val instant = LocalDateTime.of(2023, 5, 24, 12, 0).toInstant(ZoneOffset.UTC);
+        final Task task = mock(Task.class);
```

Comment:
A better approach is to use DelegateTaskFake from camunda-mockito. Then you can assert the value instead of verifying the invocation.

See https://github.com/camunda-community-hub/camunda-platform-7-mockito#delegatetaskexecutionfake

Reply:
Thank you for your feedback. So I learned a new part of the world of camunda testing. 

I updated the tests...